x-pack/plugins/cloud_security_posture/server/create_indices/create_indices.ts

```diff
@@ -4,72 +4,153 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { transformError } from '@kbn/securitysolution-es-utils';
-import { IndicesIndexSettings, MappingTypeMapping } from '@elastic/elasticsearch/lib/api/types';
 import type { ElasticsearchClient, Logger } from '@kbn/core/server';
-import { benchmarkScoreMapping } from './benchmark_score_mapping';
-import { latestFindingsMapping } from './latest_findings_mapping';
+import { errors } from '@elastic/elasticsearch';
 import {
-  LATEST_FINDINGS_INDEX_DEFAULT_NS,
-  LATEST_FINDINGS_INDEX_NAME,
   BENCHMARK_SCORE_INDEX_DEFAULT_NS,
-  BENCHMARK_SCORE_INDEX_NAME,
+  BENCHMARK_SCORE_INDEX_PATTERN,
+  BENCHMARK_SCORE_INDEX_TEMPLATE_NAME,
+  CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
   CSP_INGEST_TIMESTAMP_PIPELINE,
+  FINDINGS_INDEX_NAME,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+  LATEST_FINDINGS_INDEX_PATTERN,
+  LATEST_FINDINGS_INDEX_TEMPLATE_NAME,
 } from '../../common/constants';
 import { createPipelineIfNotExists } from './create_processor';
+import { benchmarkScoreMapping } from './benchmark_score_mapping';
 
 // TODO: Add integration tests
+
 export const initializeCspIndices = async (esClient: ElasticsearchClient, logger: Logger) => {
   await createPipelineIfNotExists(esClient, CSP_INGEST_TIMESTAMP_PIPELINE, logger);
+
   return Promise.all([
-    createIndexIfNotExists(
-      esClient,
-      LATEST_FINDINGS_INDEX_NAME,
-      LATEST_FINDINGS_INDEX_DEFAULT_NS,
-      latestFindingsMapping,
-      {},
-      logger
-    ),
-    createIndexIfNotExists(
-      esClient,
-      BENCHMARK_SCORE_INDEX_NAME,
-      BENCHMARK_SCORE_INDEX_DEFAULT_NS,
-      benchmarkScoreMapping,
-      { default_pipeline: CSP_INGEST_TIMESTAMP_PIPELINE },
-      logger
-    ),
+    createLatestFindingsIndex(esClient, logger),
+    createBenchmarkScoreIndex(esClient, logger),
   ]);
 };
 
-export const createIndexIfNotExists = async (
+const createBenchmarkScoreIndex = async (esClient: ElasticsearchClient, logger: Logger) => {
+  try {
+    const INDEX_TEMPLATE_V830 = 'cloud_security_posture.scores';
+    await deleteIndexTemplateSafe(esClient, logger, INDEX_TEMPLATE_V830);
+
+    // We always want to keep the index template updated
+    await esClient.indices.putIndexTemplate({
+      name: BENCHMARK_SCORE_INDEX_TEMPLATE_NAME,
+      index_patterns: BENCHMARK_SCORE_INDEX_PATTERN,
+      template: {
+        mappings: benchmarkScoreMapping,
+        settings: {
+          default_pipeline: CSP_INGEST_TIMESTAMP_PIPELINE,
+          lifecycle: {
+            // This is the default lifecycle name, it is named on the data-stream type (e.g, logs/ metrics)
+            name: 'logs',
+          },
+        },
+      },
+      _meta: {
+        package: {
+          name: CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+        },
+        managed_by: 'cloud_security_posture',
```

Comment:
`CLOUD_SECURITY_POSTURE_PACKAGE_NAME` ?

Reply:
So I believe it should be our plugin name - since it is managed by it (I mimicked it from an example I saw in `fleet`).
I'll create a dedicated constant for it to make that clear